language_service/src/project_system.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::LanguageService;
+use async_trait::async_trait;
+use std::{convert::Infallible, path::PathBuf};
+
+#[derive(Debug)]
+pub struct JSFileEntry {
+    pub name: String,
+    pub r#type: qsc_project::EntryType,
+}
+
+impl qsc_project::DirEntry for JSFileEntry {
+    type Error = Infallible;
+
+    fn entry_type(&self) -> Result<qsc_project::EntryType, Self::Error> {
+        Ok(self.r#type)
+    }
+
+    fn entry_extension(&self) -> String {
+        let parsed_as_path = PathBuf::from(&self.name);
+        parsed_as_path
+            .extension()
+            .map(|x| x.to_string_lossy().to_string())
+            .unwrap_or_default()
+    }
+
+    fn entry_name(&self) -> String {
+        self.name.clone()
+    }
+
+    fn path(&self) -> PathBuf {
+        PathBuf::from(&self.name)
+    }
```

Comment:
Isn't `entry_name` supposed to return just the filename portion while the `path` returns the full path including the directory query (at least that's what I see [here](https://github.com/microsoft/qsharp/blob/04e551beed2c89c9475be4bb5d6ff3907bae5039/compiler/qsc_project/src/fs.rs#L29-L35))? It seems inconsistent that these two implementations return the same data.